libcloud/test/compute/test_cloudstack.py

```diff
@@ -221,6 +221,18 @@ def test_create_node_ex_keyname(self):
         self.assertEqual(node.name, 'test')
         self.assertEqual(node.extra['key_name'], 'foobar')
 
+    def test_create_node_ex_userdata(self):
```

Comment:
Is it also possible to add a test / assert which verifies that user data is encoded correctly (you will need to do this inside the corresponding mock http class method).